tests/unit/src/test/scala/tests/codeactions/ActionableDiagnosticsSuite.scala

```diff
@@ -0,0 +1,42 @@
+package tests.codeactions
+
+import scala.meta.internal.metals.codeactions.ActionableDiagnostic
+
+import coursier.version.Version
+
+class ActionableDiagnosticsSuite
+    extends BaseCodeActionLspSuite("actionableDiagnostic") {
+
+  val oldOsLibVersion: Version = Version("0.7.8")
+
+  check(
+    "actionable-diagnostic-update",
+    s"""|//> <<>>using lib "com.lihaoyi::os-lib:${oldOsLibVersion.repr}"
```

Comment:
What's the actual diagnostic tied to this look like?

Reply:
```
[Trace - 02:46:40 PM] Received notification 'build/publishDiagnostics'
Params: {
  "textDocument": {
    "uri": "file:///private/tmp/scala-demo/Hello.scala"
  },
  "buildTarget": {
    "uri": "file:/private/tmp/scala-demo/.scala-build/?id\u003dproject_bd2c96d2de"
  },
  "diagnostics": [
    {
      "range": {
        "start": {
          "line": 0,
          "character": 15
        },
        "end": {
          "line": 0,
          "character": 40
        }
      },
      "severity": 4,
      "source": "scala-cli",
      "message": "com.lihaoyi::os-lib:0.7.8 is outdated, update to 0.8.1\n     com.lihaoyi::os-lib:0.7.8 -\u003e com.lihaoyi::os-lib:0.8.1",
      "data": {
        "range": {
          "start": {
            "line": 0,
            "character": 15
          },
          "end": {
            "line": 0,
            "character": 40
          }
        },
        "newText": "com.lihaoyi::os-lib:0.8.1"
      }
    }
  ],
  "reset": false
}
```